adapters/lifestreet.go

```diff
@@ -0,0 +1,191 @@
+package adapters
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/pbs"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+type LifestreetAdapter struct {
+	http         *HTTPAdapter
+	URI          string
+	usersyncInfo *pbs.UsersyncInfo
+}
+
+/* Name - export adapter name */
+func (a *LifestreetAdapter) Name() string {
+	return "Lifestreet"
+}
+
+// used for cookies and such
+func (a *LifestreetAdapter) FamilyName() string {
+	return "lifestreet"
+}
+
+func (a *LifestreetAdapter) GetUsersyncInfo() *pbs.UsersyncInfo {
+	return a.usersyncInfo
+}
+
+func (a *LifestreetAdapter) SkipNoCookies() bool {
+	return false
+}
+
+// parameters for Lifestreet adapter.
+type lifestreetParams struct {
+	SlotTag string `json:"slot_tag"`
+}
+
+func (a *LifestreetAdapter) callOne(ctx context.Context, req *pbs.PBSRequest, reqJSON bytes.Buffer) (result callOneResult, err error) {
+	httpReq, err := http.NewRequest("POST", a.URI, &reqJSON)
+	httpReq.Header.Add("Content-Type", "application/json;charset=utf-8")
+	httpReq.Header.Add("Accept", "application/json")
+
+	lsmResp, e := ctxhttp.Do(ctx, a.http.Client, httpReq)
+	if e != nil {
+		err = e
+		return
+	}
+
+	defer lsmResp.Body.Close()
+	body, _ := ioutil.ReadAll(lsmResp.Body)
+	result.responseBody = string(body)
+
+	result.statusCode = lsmResp.StatusCode
+
+	if lsmResp.StatusCode == 204 {
+		return
+	}
+
+	if lsmResp.StatusCode != 200 {
+		err = fmt.Errorf("HTTP status %d; body: %s", lsmResp.StatusCode, result.responseBody)
+		return
+	}
+
+	var bidResp openrtb.BidResponse
+	err = json.Unmarshal(body, &bidResp)
+	if err != nil {
+		return
+	}
+	if len(bidResp.SeatBid) == 0 {
+		return
+	}
+	if len(bidResp.SeatBid[0].Bid) == 0 {
+		return
+	}
+	bid := bidResp.SeatBid[0].Bid[0]
+
+	result.bid = &pbs.PBSBid{
+		AdUnitCode:  bid.ImpID,
+		Price:       bid.Price,
+		Adm:         bid.AdM,
+		Creative_id: bid.CrID,
+		Width:       bid.W,
+		Height:      bid.H,
+		DealId:      bid.DealID,
+		NURL:        bid.NURL,
+	}
+	return
+}
+
+func (a *LifestreetAdapter) Call(ctx context.Context, req *pbs.PBSRequest, bidder *pbs.PBSBidder) (pbs.PBSBidSlice, error) {
+	requests := make([]bytes.Buffer, len(bidder.AdUnits))
+	for i, unit := range bidder.AdUnits {
+		lsReq := makeOpenRTBGeneric(req, bidder, a.FamilyName())
```

Comment:
You'll need to update this based on latest master, the signature changed to accept mediaTypes. 
See https://github.com/prebid/prebid-server/blob/master/adapters/openrtb_util.go#L67 for the updates.

Reply:
Sounds good!